compiler/natives/src/crypto/sha256/sha256.go

```diff
@@ -0,0 +1,67 @@
+// +build js
+
+package sha256
+
+import (
+	"errors"
+
+	"github.com/gopherjs/gopherjs/js"
+)
+
+var ErrWebCryptoFailed = errors.New("crypto: Web Crypto API call failed")
```

Comment:
> Can you elaborate on how it will be used?

The idea is to have a specific error code returned when a webcrypto method fails with a `NotSupportedError`. Then most cryptographic functions implementation (in each `crypto/*` package) would look like:
```
func foo() error {
  err := webcrypto_foo()
  if err == ErrWebCryptoUnsupported {
    err = _gopherjs_overriden_foo()
  }
  return err
}
```
In order to make this error code visible from every package under `crypto`, the only way I can think of is to make it public.

> that would make user code incompatible with Go.

I didn't think about that. In the last version of the code, I moved the the non standard public stuff in a new package [crypto/subtle](https://github.com/gopherjs/gopherjs/blob/985925063ca092307686be8555800313ad780190/compiler/natives/src/crypto/subtle/subtle.go
). Is that ok? Should we put this kind of code somewhere else (like under the `js` package) ?